docs/chapter-6.mdx

```diff
@@ -106,9 +108,9 @@ import { AppState, initialState } from "./state";
 
 const _reducer = createReducer(
   initialState,
-  on(actions.addCategory, (state, action) => ({
+  on(actions.addCategory, (state, category) => ({
     ...state,
-    categories: [...state.categories, action.payload],
+    categories: [...state.categories, category],
```

Comment:
Is it intended that this snippet adds the action to `categories`? 
I would expect `category.name` here, or something similar.

Reply:
are you sure ? this just add the category passed as argument to the action in the 
```
    categories : Category[]
```
which is not a string Array

Reply:
Oh sorry, I did miss that @aboudard 
But with the current code snippet, the action's type would also be included and added to the categories array - and I think this isn't intended?

Reply:
Whoah totally ... that means we need maybe to change the signature of the action ... can I propose to be close to official doc like that :

```

export const addCategory = createAction(
  "[Category List] Add Category",
  props<{ category: Category }>()
);

on(actions.addCategory, (state, {category}) => ({
    ...state,
    categories: [...state.categories, category],
  })),

```